components/brave_wallet_ui/common/reducers/wallet_reducer.ts

```diff
@@ -117,13 +117,13 @@ reducer.on(WalletActions.setAllTokensList, (state: any, payload: GetAllTokensRet
   }
 })
 
-reducer.on(WalletActions.ethBalancesUpdated, (state: any, payload: GetBalanceReturnInfo[]) => {
+reducer.on(WalletActions.ethBalancesUpdated, (state: any, payload: GetETHBalancesPriceReturnInfo) => {
   let accounts: WalletAccountType[] = [...state.accounts]
 
   accounts.forEach((account, index) => {
-    if (payload[index].success) {
-      accounts[index].balance = payload[index].balance
-      accounts[index].fiatBalance = formatFiatBalance(payload[index].balance, 18, '2000').toString()  // TODO: use actual price info
+    if (payload.balances[index].success) {
+      accounts[index].balance = payload.balances[index].balance
+      accounts[index].fiatBalance = formatFiatBalance(payload.balances[index].balance, 18, payload.price[0].price).toString()
```

Comment:
Could we make it more explicit that we're using usd price here rather than just `payload.price[0].price`? 
Also, are other prices needed?

Reply:
It seems a little bit fragile to assume payload.price[0].price is legit and exactly what we want here.

Reply:
@Douglashdaniel Now that we only get usd price before calling this, maybe price can be changed to AssetPriceInfo type instead of `AssetPriceInfo[]`? Then we could just use the passed in price here.

Reply:
Added some changes here! Changed the type too `usdPrice` : `string`.